File: Queue1Clases.js
function Node(value){
    this.value = value;
    this.next = null;
}

function Queue(){
    this.first = null;
    this.push = QueuePush;
    this.pop = QueuePop;
    this.print = QueuePrint;
    this.getFirst = QueueGetFirst;
    this.getLast = QueueGetLast;
}
    function QueuePush(value){
        if(!this.first){
            this.first = new Node(value);
        }else{
            current = this.first;
            while(current.next){
                current=current.next;
            }
            current.next= new Node(value);
        }
    }

    function QueuePop(){
        temp = this.first;
        this.first= this.first.next;
        return temp;
    }

    function QueuePrint(){
        temp = this.first;
        while(temp.next){
            console.log(temp.value);
            temp = temp.next;
        }
        console.log(temp.value);
    }

    function QueueGetFirst(){
        return this.first;
    }

    function QueueGetLast(){
        temp = this.first;
        while(temp.next){
            temp = temp.next;
        }
        return temp;
    }

File: Unidad_1/Queue1Clases.py
class Node():
    def __init__(self,value):
        self.value = value
        self.next = None

class Queue():
    def __init__(self):
        self.first = None
    def push(self,value):
        if(not self.first):
            self.first = Node(value)
        else:
            current = self.first
            while(current.next):
                current = current.next
            current.next = Node(value)
    def pop(self):
        temp = self.first
        self.first = self.first.next
        return temp
    def getFirstValue(self):
        return self.first.value
    def getLastValue(self):
        temporalNode = self.first
        while(temporalNode.next):
            temporalNode = temporalNode.next
        return temporalNode.value
    def showAllValues(self):
        current = self.first
        while(current.next):
            print(current.value)
            current = current.next
        print(current.value)